File: Leetcode_for_beginers/Base_7/a.go
func convertToBase7(num int) string {
    needAddMinusOne := num < 0
    if num == 0 {
        return "0"
    }
    if needAddMinusOne {
        num = -1 * num
    }
    var base7s string
    for num != 0 {
        base7s = string(byte('0' + num % 7)) + base7s
        num /= 7
    }
    if needAddMinusOne {
        return "-" + base7s
    }
    return base7s
}

File: Leetcode_for_beginers/Base_7/a.java
class Solution {
    public String convertToBase7(int num) {
        if (num == 0) {
            return "0";
        }

        Boolean needAddMinusOne = num < 0;
        if (needAddMinusOne) num = -1 * num;

        StringBuilder base7s = new StringBuilder();
        while (num != 0) {
            base7s.insert(0, Character.toChars((int)'0' + num % 7));
            num /= 7;
        }
        
        if (needAddMinusOne) {
            base7s.insert(0, '-');
        }
        return base7s.toString();
    }
}